nx/test/nx/tensor_test.exs

```diff
@@ -127,4 +127,69 @@ defmodule Nx.TensorTest do
       assert_raise RuntimeError, "not supported", fn -> Nx.random_uniform({2, 2}) end
     end
   end
+
+  describe "access with :all" do
+    test "should access the entire tensor" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[:all]] == t
+      assert t[[:all, :all]] == t
+    end
+
+    test "should take the first column" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[:all, 0]] == Nx.tensor([1, 4, 7])
+    end
+
+    test "should take the second column" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[:all, 1]] == Nx.tensor([2, 5, 8])
+    end
+
+    test "should take the first 2 columns" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[:all, 0..1]] == Nx.tensor([[1, 2], [4, 5], [7, 8]])
+    end
+
+    test "should take the last 2 columns" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[:all, 1..2]] == Nx.tensor([[2, 3], [5, 6], [8, 9]])
+    end
+
+    test "should take the first row" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[0, :all]] == Nx.tensor([1, 2, 3])
+    end
+
+    test "should take the second row" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[1, :all]] == Nx.tensor([4, 5, 6])
+    end
+
+    test "should take the first two rows" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[0..1, :all]] == Nx.tensor([[1, 2, 3], [4, 5, 6]])
+    end
+
+    test "should take the last two rows" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[1..2, :all]] == Nx.tensor([[4, 5, 6], [7, 8, 9]])
+    end
+
+    test "should raise when out of bounds" do
```

Comment:
I think we should have a higher dimensional test as well, like a 3D tensor having :all along each o the axes